frontend/src/employee-frontend/components/child-information/ChildDiscussions.tsx

```diff
@@ -0,0 +1,224 @@
+  display: flex;
+  min-height: 2rem;
+`
+
+const DataLabel = styled.div`
+  width: 240px;
+  padding: 0 40px 0 0;
+  margin: 0;
+  font-weight: ${fontWeights.semibold};
+`
+
+const DataValue = styled.div`
+  display: flex;
+`
+
+const RightAlignedRow = styled.div`
+  display: flex;
+  align-items: baseline;
+  justify-content: flex-end;
+  padding-top: 20px;
+`
+
+export default React.memo(function ChildDiscussions({ childId }: Props) {
+  const { i18n, lang } = useTranslation()
+
+  const [offeredDate, setOfferedDate] = useState<LocalDate | null>(null)
+  const [heldDate, setHeldDate] = useState<LocalDate | null>(null)
+  const [counselingDate, setCounselingDate] = useState<LocalDate | null>(null)
+
+  const childDiscussionData = useQueryResult(childDiscussionQuery(childId))
+
+  useEffect(() => {
+    if (!childDiscussionData.isSuccess) return
+
+    setOfferedDate(childDiscussionData.value.offeredDate)
+    setHeldDate(childDiscussionData.value.heldDate)
+    setCounselingDate(childDiscussionData.value.counselingDate)
+  }, [childDiscussionData])
+
+  const { mutateAsync: createChildDiscussion } = useMutationResult(
+    createChildDiscussionMutation
+  )
+  const { mutateAsync: updateChildDiscussion } = useMutationResult(
+    updateChildDiscussionMutation
+  )
+
+  const { uiMode, toggleUiMode, clearUiMode } = useContext<UiState>(UIContext)
+  const editing = uiMode == 'child-discussion-editing'
+
+  const startEdit = useCallback(() => {
+    if (childDiscussionData.isSuccess) {
+      setOfferedDate(childDiscussionData.value.offeredDate)
+      setHeldDate(childDiscussionData.value.heldDate)
+      setCounselingDate(childDiscussionData.value.counselingDate)
+    }
+    toggleUiMode('child-discussion-editing')
+  }, [childDiscussionData, toggleUiMode])
+
+  const handleCancel = () => {
+    if (childDiscussionData.isSuccess) {
+      setOfferedDate(childDiscussionData.value.offeredDate)
+      setHeldDate(childDiscussionData.value.heldDate)
+      setCounselingDate(childDiscussionData.value.counselingDate)
+    }
+    clearUiMode()
+  }
+
+  const handleSaveChildDiscussion = () => {
+    if (childDiscussionData.isSuccess && childDiscussionData.value.id) {
+      return updateChildDiscussion({
+        childId,
+        data: {
+          offeredDate,
+          heldDate,
+          counselingDate
+        }
+      })
+    } else {
+      return createChildDiscussion({
+        childId,
+        data: {
+          offeredDate,
+          heldDate,
+          counselingDate
+        }
+      })
+    }
+  }
+
+  const handleSuccess = () => {
+    clearUiMode()
+  }
+
+  return (
+    <div>
+      <FlexRow justifyContent="space-between">
+        <Title size={4}>{i18n.childInformation.childDiscussion.title}</Title>
+        {!editing && (
+          <RequireRole
+            oneOf={[
+              'UNIT_SUPERVISOR',
+              'ADMIN',
+              'STAFF',
+              'SPECIAL_EDUCATION_TEACHER'
+            ]}
+          >
```

Comment:
Hard-coded role lists should not be used in new frontend code, and you should use the `permittedAction` mechanism instead. The `/children/{childId}` route returns actions that are permitted on the given child, and you should check here whether it contains the `CREATE_CHILD_DISCUSSION` action.

Reply:
I see, well that sounds like a better way. I saw this hard-coded way somewhere and just used it here also. I'll add the changes to use permittedAction to check the required actions.